packages/flutter/lib/src/widgets/scroll_delegate.dart

```diff
@@ -413,6 +413,60 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// none of the children will ever try to keep themselves alive.
   ///
   /// Defaults to true.
+  ///
+  /// {@tool dartpad}
+  /// This sample demonstrates how to use the [AutomaticKeepAlive] widget in
+  /// combination with the [AutomaticKeepAliveClientMixin] to selectively preserve
+  /// the state of individual items in a scrollable list.
+  ///
+  /// Normally, widgets in a lazily built list like [ListView.builder] are
+  /// disposed of when they leave the visible area to save resources. This means
+  /// that any state inside a [StatefulWidget] would be lost unless explicitly
+  /// preserved.
+  ///
+  /// In this example, each list item is a [StatefulWidget] that includes a
+  /// counter and an increment button. To preserve the state of selected items
+  /// (based on their index), the [AutomaticKeepAlive] widget and
+  /// [AutomaticKeepAliveClientMixin] are used:
+  ///
+  /// - The `wantKeepAlive` getter in the item’s state class returns true for
+  ///   even-indexed items, indicating that their state should be preserved.
+  /// - For odd-indexed items, `wantKeepAlive` returns false, so their state is
+  ///   not preserved when scrolled out of view.
```

Comment:
Those 2 code examples are copied from the documentation of `AutomaticKeepAlive` and `KeepAlive`. Is there a better way than duplicating them ?

Reply:
I think you can insert a template inside of another template. .. or put a `@macro` (use of template), inside of a `@template`